src/mod_muc_room.erl

```diff
@@ -883,22 +893,31 @@ broadcast_room_packet(From, FromNick, Role, Packet, StateData) ->
             RouteFrom = jid:replace_resource(StateData#state.jid,
                                              FromNick),
             RoomJid = StateData#state.jid,
-
-            ejabberd_hooks:run(update_inbox_for_muc, StateData#state.server_host,
-                               [RoomJid, {From, RouteFrom}, StateData#state.affiliations, FilteredPacket]),
-
-            lists:foreach(fun({_LJID, Info}) ->
+            HookInfo = #{room_jid => RoomJid,
+                         from_jid => From,
+                         from_room_jid => RouteFrom,
+                         packet => FilteredPacket,
+                         affiliations_map => StateData#state.affiliations},
+            run_update_inbox_for_muc_hook(StateData#state.server_host, HookInfo),
```

Comment:
Have you considered merging this hook with `room_send_packet`?

Reply:
yeah, but muclight does not have that affiliations map.